test/e2e_federation/replicaset.go

```diff
@@ -369,6 +369,11 @@ func createReplicaSetOrFail(clientset *fedclientset.Clientset, replicaset *v1bet
 	}
 	By(fmt.Sprintf("Creating federation replicaset %q in namespace %q", replicaset.Name, namespace))
 
+	// a hack to pass validation check for controller selector immutability prior create (technically an update to nil object)
+	consistentLabelMap := map[string]string{"name": "myrs-consistent"}
+	replicaset.Spec.Selector.MatchLabels = consistentLabelMap
+	replicaset.Spec.Template.ObjectMeta.Labels = consistentLabelMap
```

Comment:
@liggitt  This hack is needed as a new selector is generated every time creating a new RS for updating (and creating) purpose. Refer this for more information.

Aside from this, there are two ways to resolve this:
(1) make the affected tests run in serial mode (drawback: slow)
(2) create a constant and pass it as function parameter

WDYT?